compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/Compiler.java

```diff
@@ -164,7 +164,9 @@ public void list(String sourcePackage) {
         for (BLangPackage pkgNode : packages) {
             if (pkgNode.symbol != null) {
                 this.compilerDriver.compilePackage(pkgNode);
-                dlog.resetErrorCount();
+                if (this.dlog.getErrorCount() > 0) {
+                    throw new BLangCompilerException("compilation contains errors");
```

Comment:
Can you share the output we get for this error. We would need to provide more info for the user to understand where it fails.